src/math/math_base.h

```diff
@@ -105,5 +105,81 @@ struct PointBase3D : public PointBase2D<_Type>
     _Type z;
 };
 
+template <typename _Type>
+struct LineBase2D
+{
+    LineBase2D( const PointBase2D<_Type> & _p1, const PointBase2D<_Type> & _p2 )
+        : p1( _p1 )
+        , p2( _p2 )
+    { }
+
+    LineBase2D( const PointBase2D<_Type> & _p, _Type angleRadians )
+        : p1( _p )
+        , p2( _p.x + std::cos(angleRadians), _p.y + std::sin(angleRadians) )
+    { }
+
+    bool operator == ( const LineBase2D & line ) const
+    {
+        return pvmath::isEqual<_Type>(angle(), line.angle()) && pvmath::isEqual<_Type>(distance(line.p1), 0);
```

Comment:
Actually, angle approach won't work for all cases. Let's assume we have a line with points A and B (for p1 and p2) and line with points B and A. Angles would be opposite. To solve this problem we need to have `parallel` function:
```cpp
return parallel( line ) && pvmath::isEqual<_Type>(distance(line.p1), 0);
```
then it's definitely the same line.